kcidb/oo/__init__.py

```diff
@@ -187,6 +188,39 @@ def architecture_valid_builds(self):
             reverse=True
         ))
 
+    def get_build_failure_log(self, log_excerpt):
+        """Get one liner log for build failure"""
+        log_list = log_excerpt.split("\n")
+
+        patterns = [
+            'Kernel panic',
+            't access tty',
+            r'.*\.c:.*error.*',
+            r'.*\.h:.*error.*',
+            r'.*error.*modpost',
+            'No rule to make target',
+            'tail will be killed now',
+            r'.*error.*',
+            r'.*No such file.*',
+            r'kern.*No irq handler for vector',
+            'package_dtbs=done',
+            'RESULT=fail'
+        ]
+
+        for pattern in patterns:
+            for log in log_list:
+                if re.search(pattern, log):
+                    return log
+
+        # If pattern is not found return last non-empty log line
+        log_list.reverse()
+        for log in log_list:
+            log = log.strip()
+            if log:
+                return log
```

Comment:
If you don't feel comfortable messing around with those regexes, splitting lines and doing loops is alright too. The memory won't be an issue until we get to processing the full logs, and your approach is quite clear. 👍🏻

Reply:
Btw in some cases logs might be enormous, but i guess unlikely they will reach kcidb with such size.
Also this patterns need to be well-though, maybe even not hardcoded, but separate (config?) file, as they might need to be updated often.

Reply:
> If you don't feel comfortable messing around with those regexes, splitting lines and doing loops is alright too. The memory won't be an issue until we get to processing the full logs, and your approach is quite clear. 👍🏻

Yes, I would rather keep it simpler.

Reply:
> Btw in some cases logs might be enormous, but i guess unlikely they will reach kcidb with such size.
Also this patterns need to be well-though, maybe even not hardcoded, but separate (config?) file, as they might need to be updated often.

We would have to limit the size of files we archive, with the particular limit yet to be decided. Otherwise, yes, I think we should process only the last N MB of each log file for performance reasons.